spirv_glsl.cpp

```diff
@@ -333,7 +336,8 @@ void CompilerGLSL::reset()
 
 	ir.for_each_typed_id<SPIRVariable>([&](uint32_t, SPIRVariable &var) { var.dependees.clear(); });
 
-	ir.reset_all_of_type<SPIRExpression>();
+    // Track write-throughs for loop variables - dxc likes to generate them
```

Comment:
What does this mean?

Reply:
Some of those changes might be solved elsewhere now. Updating all the test shaders with this changed being reverted didn't alter any of the reference shaders, so this might now longer be needed.
I'm preparing all necessary changes and submit it when I'm done.